### ISSUE TITLE
(Variable not shown in template UI) Templates when using Traefik

### ISSUE THREAD

**Issue by flololan** [portainer/templates#121]
Hi everyone. 

I have a Docker Sarm (3 nodes, 2 Workers & 1 Manager). With portainer.

Everything is working fine but I'm trying to add some templates now. 
As far as I was able to test, it is crucial that the name of the service is only allowed to exist once to not collide with another. So I tried having a variable that you have to put in when using templates. Here one line as example:
```traefik.http.routers.${NAMEOFSERVICE}.rule=Host(`${HOSTNAME}`)```

When using the template I have the field for the hostname. But not for the service name. How do I do that?

Templates.json:
```{
    "type": 2,
    "title": "Hello World test",
    "description": "Kleiner Service/Stack zum Testen der Konfiguration",
    "note": "Mit aktiviertem TLS",
    "categories": ["Tools"],
    "platform": "linux",
    "logo": "http://www.digitalconnectmag.com/wp-content/uploads/2016/01/tutum.png",
    "repository": {
      "url": "https://LOCALGITREPO/portainer-templates",
      "stackfile": "helloworld.yml"
    },
    "env": [
      {
        "name": "HOSTNAME",
        "label": "Domain/Hostname",
        "description": "Der Dateipfad für die Traefik-Config (dort muss die certificates.toml-Datei liegen). Beispiel: /traefik/configuration/ "
      },
	  {
        "name": "NAMEOFSERVICE",
        "label": "Name des Services (Muss eindeutig sein!)",
        "description": "Darf es nicht zwei mal geben, da sonst das Routing für beide nicht mehr funktioniert!"
      }
    ]
  }
```

I do not see what I've might done wrong.
Thank you in advance for any help...

![image](https://user-images.githubusercontent.com/12832544/82198078-146ba180-98fc-11ea-862d-247e38e11f81.png)

**Comment by flololan** [comment]
Also posted here: https://forums.portainer.io/t/templates-when-using-traefik/464

**Comment by deviantony** [comment]
Hi @flololan 

If I understand correctly then the "traefik.http.routers.${NAMEOFSERVICE}.rule=Host(`${HOSTNAME}`)" is located inside the `helloworld.yml` ?

Everything defined in the `env` property of the template will be passed to the environment (via `export`) when deploying a stack file, you'll want to make sure your setup works properly with the CLI first.

Something like:

```
export HOSTNAME=value1
export NAMEOFSERVICE=value2
docker stack deploy -c helloworld.yaml stackname
```

**Comment by flololan** [comment]
Thanks for that!

I tried exactly that and while the service/stack started without issue, it seems that Traefik couldn't route properly to it. Logs showed something like:

> Router defined multiple times with different configurations

Solution 

Instead of (in helloworld.yml)
```traefik.http.routers.${NAMEOFSERVICE}.rule=Host(`${HOSTNAME}`)```
I had to add something like "helloworld-"bevore the variable.
```traefik.http.routers.helloworld-${NAMEOFSERVICE}.rule=Host(`${HOSTNAME}`)```

With this change the field NAMEOFSERVICE is being shown in the Templates-UI and the services/stacks run without any issues (so far).
